cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c

```diff
@@ -337,59 +337,83 @@ img_mgmt_impl_read(int slot, unsigned int offset, void *dst,
     return 0;
 }
 
+/*
+ * The alloc_ctx and free_ctx are specifically provided for
+ * the img_mgmt_impl_write_image_data to allocate/free single flash_img_context
+ * type buffer.
+ * When heap is enabled these functions will operate on heap; when  heap is not
+ * allocated the alloc_ctx just returns pointer to static, global life-time
+ * variable, and free_ctx does nothing.
+ * CONFIG_HEAP_MEM_POOL_SIZE is C preprocessor literal.
+ */
+static inline struct flash_img_context *alloc_ctx()
```

Comment:
I would add the `void` parameter here.

Reply:
Any particular reason for this and `free_ctx` being `inline`?

Reply:
Will add void.
Regarding inline: these functions have been crated only for the `img_mgmt_impl_write_image_data` and each is called once; bodies of these functions are close to nothing (the  `CONFIG_HEAP_MEM_POOL_SIZE` is literal and will "optimize" one of paths out), so I have figured out that making all the call effort is pointless, just inline it.